staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -371,7 +372,7 @@ func summarizeData(data []byte, maxLength int) string {
 	}
 }
 
-func limitedReadBody(req *http.Request, limit int64) ([]byte, error) {
+func limitedReadBody(ctx context.Context, req *http.Request, limit int64, resourceGroup string, verb string) ([]byte, error) {
```

Comment:
Are you not interested in logging the size of what was read if the the limit was 0 or less (line 378 below).

Reply:
I don't know if the function signature is right. It looks odd to me that you have to take all these new extra parameters, I see this as a smell that this function now has too many roles. Can you keep this method as it is and possibly make a helper function log the size of the returned data?

Reply:
> Are you not interested in logging the size of what was read if the the limit was 0 or less (line 378 below).

We're not reading anything at that point. 

> Can you keep this method as it is and possibly make a helper function log the size of the returned data?

Sure.
How about I

Reply:
Yeah but it's using limitedReader...

```
func (l *LimitedReader) Read(p []byte) (n int, err error) {
	if l.N <= 0 {
		return 0, EOF
	}
	if int64(len(p)) > l.N {
		p = p[0:l.N]
	}
	n, err = l.R.Read(p)
	l.N -= int64(n)
	return
}
```